include/votca/xtp/threecenter.h

```diff
@@ -102,10 +110,25 @@ class TCMatrix_gwbse : public TCMatrix {
 
   void MultiplyRightWithAuxMatrix(const Eigen::MatrixXd& AuxMatrix);
 
+  void MultiplyRightWithAuxMatrixOpenMP(const Eigen::MatrixXd& AuxMatrix);
+
+  void TripleTensorProduct(std::vector<Eigen::MatrixXd>& block,
+                           const std::vector<Eigen::MatrixXd>& symmstorage,
+                           const Eigen::MatrixXd& dftn,
+                           const Eigen::MatrixXd& dftm, int numFunc) const;
+
  private:
   // store vector of matrices
   std::vector<Eigen::MatrixXd> _matrix;
 
+  // Logger
+  Logger& _log;
+
+  // Use GPU if available
+#if defined(USE_GPU)
```

Comment:
Because If a move it to the part of the code that calls the tensor matrix multiplication, it results in multiple calls to `cublasCreate(& _handle)` that is an expensive operation.

Reply:
how expensive is it? I would assume in comparison to the whole tensor multiplication it is pretty cheap.